moz_ingest/io_modules/decoders/moz_ingest/common.lua

```diff
@@ -291,7 +321,7 @@ function transform_message(hsr, msg)
         if not ip_addr then
             ip_addr = hsr:read_message("Fields[remote_addr]")
         end
-        msg.Fields.geoCity, msg.Fields.geoCountry = get_geo_city(ip_addr)
+        msg.Fields.geoCity, msg.Fields.geoSubdivision1, msg.Fields.geoSubdivision2, msg.Fields.geoCountry = get_geo_city(ip_addr)
```

Comment:
We should leverage the Field array support here and avoid a breaking data change later.  This allows the use of single field of 'geoSubdivisions'. It should not be an issue for https://bugzilla.mozilla.org/show_bug.cgi?id=1447652#c7 since the immediate use case (US state) only has a single value. Is there a bug to fix the python/scala protobuf readers?

Reply:
I was thinking about the python/scala readers, and I don't see a straightforward way to fix them. The difficulty is that there doesn't seem to be a way to distinguish between a scalar field with one item vs. an array field with one item. As such, we would either have to add special cases for fields we expect to be arrays (gross) or support fields that can be either a single scalar item or an array of items, which would introduce complexity in consumers of the data.

Reply:
If we change the schema in the future (rename the fields) all down stream consumer would have to change.

Reply:
Regarding reading the protobuf?  That is pretty standard, so I take it the auto-generated APIs from the protobuf definition are not being used?

Reply:
The protobuf definitions are being used, but there is a convenience layer on top of that to allow access to the pings in (roughly) their original JSON structure. This layer includes fields added during decoding in a "metadata" section (similar to how the direct-to-parquet outputs are set up). There's no specification for data types for these fields, so they are all assumed to be scalars (using the first available value of the indicated type).